lib/krikri/harvesters/marc_xml_harvester.rb

```diff
@@ -0,0 +1,106 @@
+module Krikri::Harvesters
+
+  class MarcXMLHarvester
```

Comment:
Can we add some class level docs for implementation requirements?  

Seems like a subclass will have to implement `#each_collection`, so maybe this should ship with an explicitly abstract method throwing `NotImplementedError`. That is the pattern we have followed for abstract methods to date.